src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -764,17 +801,14 @@ def renew_certs(self, job):
     # CERTIFICATE_CREATE_CSR          - __create_csr
     # CERTIFICATE_CREATE_ACME         - __create_acme_certificate
 
-    # TODO: Make the following method inaccessible publicly
-    # CERTIFICATE_CREATE              - __create_certificate ( ONLY TO BE USED INTERNALLY )
-
     @accepts(
         Dict(
             'certificate_create',
             Bool('tos'),
             Dict('dns_mapping', additional_attrs=True),
             Int('csr_id'),
             Int('signedby'),
-            Int('key_length'),
+            Int('key_length', enum=[1024, 2048, 4096]),
```

Comment:
Dont we allow ECDSA with less than 1024?

Reply:
This field applies to when certs are being created. We don't support creating ec keys at all ( it is only allowed during import ). I'll have a ticket for this where we add support for ec keys during creation time of a cert where a user can specify if he/she wants to use rsa or an ec based key with the selected curve